### ISSUE TITLE
updating from 24.0 to 24.1

### ISSUE THREAD

**Issue by alpapan** [galaxyproject/galaxy#18800]
**Describe the bug**

i updated from 23.2 to 24.0 without issues and then trying to update from 24.0 to 24.1. I used the instructions in the release notes.

1. Database must be updated with `./manage_db.sh upgrade` but this is not done automatically. 
I had to run it twice (./run.sh; ./manage_db.sh upgrade; ./run.sh ; ./manage_db.sh upgrade; ./run.sh) to get it of the error.

2. the [documentation](https://docs.galaxyproject.org/en/master/releases/24.1_announce.html) asks us to uninstall fastapi but then receive this error

```
git checkout release_24.1 && git pull --ff-only origin release_24.1
. ./venv/bin/activate
pip uninstall fastapi
./run.sh
```

```
DEBUG:yacman.yacman_future:Using yacman version 0.9.3
Traceback (most recent call last):
  File "/mnt/lazebnik/home/galaxy/.venv/bin/gunicorn", line 8, in <module>
    sys.exit(run())
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/app/wsgiapp.py", line 67, in run
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]", prog=prog).run()
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/app/base.py", line 236, in run
    super().run()
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/app/base.py", line 72, in run
    Arbiter(self).run()
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 58, in __init__
    self.setup(app)
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/arbiter.py", line 118, in setup
    self.app.wsgi()
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/mnt/lazebnik/home/galaxy/.venv/lib/python3.8/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/webapps/galaxy/fast_factory.py", line 50, in <module>
    from .fast_app import initialize_fast_app
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/webapps/galaxy/fast_app.py", line 7, in <module>
    from fastapi import (
ModuleNotFoundError: No module named 'fastapi'
```

3. it keeps expecting interactive tools but i've turned them off. so i get these errors, no idea where they are coming from
config has `interactivetools_enable: false`

```
AssertionError: Trying to load an InteractiveTool, but InteractiveTools are not enabled.
galaxy.tool_util.toolbox.base ERROR 2024-09-11 19:00:22,454 [pN:handler0,p:1649898,tN:MainThread] Error reading tool from path: interactivetool_two_entry_points.xml
Traceback (most recent call last):
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/tool_util/toolbox/base.py", line 881, in _load_tool_tag_set
    tool = self.load_tool(concrete_path, use_cached=False, tool_cache_data_dir=tool_cache_data_dir)
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/tool_util/toolbox/base.py", line 1108, in load_tool
    tool = self.create_tool(
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/tools/__init__.py", line 548, in create_tool
    return self._create_tool_from_source(tool_source, config_file=config_file, **kwds)
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/tools/__init__.py", line 564, in _create_tool_from_source
    return create_tool_from_source(self.app, tool_source, **kwds)
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/tools/__init__.py", line 356, in create_tool_from_source
    tool = ToolClass(config_file, tool_source, app, **kwds)
  File "/mnt/lazebnik/home/galaxy/lib/galaxy/tools/__init__.py", line 3102, in __init__
    assert app.config.interactivetools_enable, ValueError(
AssertionError: Trying to load an InteractiveTool, but InteractiveTools are not enabled.
```

So :
1. is purely FYI
2. Is fastapi needed or not?
3. how do i avoid these interactivetool assertion errors?

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 24.2
Commit: a5da8acfc581c77de8d5c5b460ee8e433c295c6e

**Comment by alpapan** [comment]
[err.txt](https://github.com/user-attachments/files/16961105/err.txt)

**Comment by mvdbeek** [comment]
1. fastapi-slim also needs to be re-installed if you didn't uninstall fastapi before the upgrade, as that's a namespace package that installs into `site-packages/fastapi`. if you remove `fastapi` you also remove the files that `fastapi-slim` installed, while pip thinks it still is installed. I'll update the instructions.
2. Do you have the output of the first migration ? There shouldn't be a need to run the migration twice if it didn't fail the first time.
3. Your tool config file loads interactive tools. If you use the sample, copy the sample file to `config/tool_conf.xml` and remove the interactive tools.

**Comment by alpapan** [comment]
Yeap. I had uninstalled fastapi before the upgrade, i followed the instructions.

2. the output of first migration was identical to second... no idea why it happened like that. Update to 24.0 was fine.

3. i have no interactive tools. It somehow wants to use the test/ ones (which I think got downloaded when i followed the 24.0 instructions of `git checkout release_24.0 && git pull --ff-only origin release_24.0`

to clarify. My 23.2 works fine. The upgrade to 24.0 was fine. No config was changed to upgrade to 24.2 and i followed the instructions as per my message above: 
```
git checkout release_24.1 && git pull --ff-only origin release_24.1
. ./venv/bin/activate
pip uninstall fastapi
./run.sh
```
